pkg/bgpv1/manager/reconcilerv2/pod_cidr.go

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/instance"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+)
+
+type ExportPodCIDRReconcilerOut struct {
+	cell.Out
+
+	Reconciler ConfigReconciler `group:"bgp-config-reconciler-v2"`
+}
+
+type ExportPodCIDRReconcilerIn struct {
+	cell.In
+
+	Logger     logrus.FieldLogger
+	PeerAdvert *CiliumPeerAdvertisement
+}
+
+type ExportPodCIDRReconciler struct {
+	logger     logrus.FieldLogger
+	peerAdvert *CiliumPeerAdvertisement
+}
+
+// ExportPodCIDRMetadata is a map of advertisements per family, key is family type
+type ExportPodCIDRMetadata map[types.Family][]*types.Path
+
+func NewExportPodCIDRReconciler(params ExportPodCIDRReconcilerIn) ExportPodCIDRReconcilerOut {
+	return ExportPodCIDRReconcilerOut{
+		Reconciler: &ExportPodCIDRReconciler{
+			logger:     params.Logger.WithField(types.ReconcilerLogField, "PodCIDR"),
+			peerAdvert: params.PeerAdvert,
+		},
+	}
+}
+
+func (r *ExportPodCIDRReconciler) Name() string {
+	return "ExportPodCIDR"
+}
+
+func (r *ExportPodCIDRReconciler) Priority() int {
+	return 30
+}
+
+func (r *ExportPodCIDRReconciler) Reconcile(ctx context.Context, p ReconcileParams) error {
+	if p.DesiredConfig == nil {
+		return fmt.Errorf("BUG: ExportPodCIDR reconciler called with nil CiliumBGPNodeConfig")
+	}
+
+	if p.CiliumNode == nil {
+		return fmt.Errorf("BUG: ExportPodCIDR reconciler called with nil CiliumNode")
+	}
+
+	// get pod CIDR prefixes
+	var podCIDRPrefixes []netip.Prefix
+	for _, cidr := range p.CiliumNode.Spec.IPAM.PodCIDRs {
+		prefix, err := netip.ParsePrefix(cidr)
+		if err != nil {
+			return fmt.Errorf("failed to parse prefix %s: %w", cidr, err)
+		}
+		podCIDRPrefixes = append(podCIDRPrefixes, prefix)
+	}
+
+	// get desired paths per address family
+	desiredFamilyAdverts, err := r.getDesiredPathsPerFamily(p, podCIDRPrefixes)
+	if err != nil {
+		return err
+	}
+
+	return r.reconcileFamilyAdvertisements(ctx, p, desiredFamilyAdverts)
+}
+
+// getDesiredPathsPerFamily returns a map of desired paths per address family.
+// Note: This returns prefixes per address family. Global routing table will contain prefix per family not per neighbor.
+// Per neighbor advertisement will be controlled by BGP Policy.
```

Comment:
It will be added in subsequent PR - one where we add policy. By default, we will then disable advertisements and only advertise based on the peer policy action.